tests/unit/type/stream.tcl

```diff
@@ -49,6 +49,23 @@ proc streamSimulateXRANGE {items start end} {
 
 set content {} ;# Will be populated with Tcl side copy of the stream content.
 
+proc stream_response_interpreter {id response} {
+    if {!$::force_resp3 || $::redis::testing_resp3($id) == 1} {
+        return $response
+    }
+    if  {[string is list $response] && ([llength $response]&1) == 0 && [llength [lindex $response 0]] ne 2} {
+        set tuparray {}
+        foreach {key val} $response {
+            set tmp {}
+            lappend tmp $key
+            lappend tmp $val
+            lappend tuparray $tmp
+        }
+        return $tuparray
+    }
+    return $response
+}
```

Comment:
lets do what we said on the call, and add a mechanism for implicit translators in redis.tcl (or another file next to it), so that we don't have to wrap each new call to XREAD that's added.